Loom 0.11 subproject on Mac: Could not find java-objc-bridge-1.0.0-natives-osx.jar

Hello, I've run into an issue setting up the MultiLoader template and they suggested this might be an issue with Loom itself.
This does not occur with the standard Fabric setup from FabricMC/fabric-example-mod, so I suspect either something is off with the multiloader setup or the multiloader setup hits some edge case that the example mod does not. Changing the Fabric/build.gradle to use Loom 0.10-SNAPSHOT does not have the issue, so it seems likely that some change happened to introduce this problem with Loom 0.11.

[root build.gradle](https://github.com/jaredlll08/MultiLoader-Template/blob/1.18/build.gradle)
[fabric build.gradle](https://github.com/jaredlll08/MultiLoader-Template/blob/1.18/Fabric/build.gradle)

Specifically, when trying to gradle sync in Idea, I encounter the following error:
```
Starting Gradle Daemon...
Gradle Daemon started in 1 s 446 ms

> Configure project :Common
SpongePowered Vanilla 'GRADLE' Toolset Version '0.2.1-SNAPSHOT'
Targeting Minecraft '1.18.2' on a 'JOINED' platform

> Configure project :Fabric
Fabric Loom: 0.11.32

> Configure project :Forge
WARNING: This project is configured to use the official obfuscation mappings provided by Mojang. These mapping fall under their associated license, you should be fully aware of this license. For the latest license text, refer below, or the reference copy here: https://github.com/MinecraftForge/MCPConfig/blob/master/Mojang.md, You can hide this warning by running the `hideOfficialWarningUntilChanged` task
WARNING: (c) 2020 Microsoft Corporation. These mappings are provided "as-is" and you bear the risk of using them. You may copy and use the mappings for development purposes, but you may not redistribute the mappings complete and unmodified. Microsoft makes no warranties, express or implied, with respect to the mappings provided here.  Use and modification of this document or the source code (in any form) of Minecraft: Java Edition is governed by the Minecraft End User License Agreement available at https://account.mojang.com/documents/minecraft_eula.

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':Fabric:configureClientLaunch'.
> Could not create task ':Fabric:extractNatives'.
   > Could not create task of type 'ExtractNativesTask'.
      > Could not resolve all files for configuration ':Fabric:minecraftNatives'.
         > Could not find java-objc-bridge-1.0.0-natives-osx.jar (ca.weblite:java-objc-bridge:1.0.0).
           Searched in the following locations:
               https://repo.maven.apache.org/maven2/ca/weblite/java-objc-bridge/1.0.0/java-objc-bridge-1.0.0-natives-osx.jar

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 14s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.4.2/userguide/command_line_interface.html#sec:command_line_warnings
```

jaredlll08/MultiLoader-Template#11

Let me know any suggestions or if you need any additional details.

--- a/src/main/java/net/fabricmc/loom/LoomRepositoryPlugin.java
+++ b/src/main/java/net/fabricmc/loom/LoomRepositoryPlugin.java
@@ -26,7 +26,9 @@ package net.fabricmc.loom;
 
 import org.gradle.api.Plugin;
 import org.gradle.api.Project;
+import org.gradle.api.artifacts.ArtifactRepositoryContainer;
 import org.gradle.api.artifacts.dsl.RepositoryHandler;
+import org.gradle.api.artifacts.repositories.ArtifactRepository;
 import org.gradle.api.artifacts.repositories.IvyArtifactRepository;
 import org.gradle.api.artifacts.repositories.MavenArtifactRepository;
 import org.gradle.api.initialization.Settings;
@@ -68,7 +70,8 @@ public class LoomRepositoryPlugin implements Plugin<PluginAware> {
 			repo.setName("Fabric");
 			repo.setUrl(MirrorUtil.getFabricRepository(target));
 		});
-		repositories.maven(repo -> {
+
+		MavenArtifactRepository mojangRepo = repositories.maven(repo -> {
 			repo.setName("Mojang");
 			repo.setUrl(MirrorUtil.getLibrariesBase(target));
 
@@ -79,6 +82,16 @@ public class LoomRepositoryPlugin implements Plugin<PluginAware> {
 				sources.ignoreGradleMetadataRedirection();
 			});
 		});
+
+		// If a mavenCentral repo is already defined, remove the mojang repo and add it back before the mavenCentral repo so that it will be checked first.
+		// See: https://github.com/FabricMC/fabric-loom/issues/621
+		ArtifactRepository mavenCentral = repositories.findByName(ArtifactRepositoryContainer.DEFAULT_MAVEN_CENTRAL_REPO_NAME);
+
+		if (mavenCentral != null) {
+			repositories.remove(mojangRepo);
+			repositories.add(repositories.indexOf(mavenCentral), mojangRepo);
+		}
+
 		repositories.mavenCentral();
 
 		repositories.ivy(repo -> {

